src/huggingface_hub/utils/_errors.py

```diff
@@ -0,0 +1,41 @@
+from requests import HTTPError
+
+
+class RepositoryNotFoundError(HTTPError):
+    """
+    Raised when trying to access a hf.co URL with an invalid repository name, or
+    with a private repo name the user does not have access to.
+    """
```

Comment:
The first line should be a short single line explanation, then a longer description if necessary, an an example of when the exception would be raised would also be nice.

Reply:
Good call! Updated in latest commit